src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -20,20 +20,31 @@ internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedO
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioExperimentationService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
-            : base(threadingContext, assertIsForeground: true)
+            : base(threadingContext)
         {
-            try
+            object experimentationServiceOpt = null;
+            MethodInfo isCachedFlightEnabledInfo = null;
+
+            threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
This is isn't fire-and-forget, right, so this constructor will block on the background thread if the UI thread is blocked? It would seem if we were initializing this on a background thread, we might just be converting this from a crash to a deadlock if the UI thread wasn't pumping when we hit the crashes?

I could understand this might be a tactical fix and that's a risk we're willing to take, but I'd want that very clearly spelled out if it was.

Reply:
> It would seem if we were initializing this on a background thread, we might just be converting this from a crash to a deadlock if the UI thread wasn't pumping when we hit the crashes?

The UI thread doesn't block JTF operations. A deadlock can only occur if someone is failing to follow the rules of the application.

Reply:
> The UI thread doesn't block JTF operations  A deadlock can only occur if someone is failing to follow the rules of the application.

How do we know Roslyn isn't doing a .Result somewhere? Isn't all the code that's calling this today not following JTF rules?

Reply:
I verified locally that 28307.51.d16.0 reproduces this bug, and that the bug does not occur with this change applied.

Reply:
Sounds good, thanks for doing some testing.